src/Command/SiteDbImportCommand.php

```diff
@@ -228,4 +247,56 @@ protected function execute(InputInterface $input, OutputInterface $output) {
     }
 
   }
+
+  /**
+   * Downloads the dump from s3 bucket.
+   *
+   * @param $filename
+   *
+   * @return string The absolute path for the dump.
+   *
+   * @throws SiteCommandException
+   */
+  protected function download($filename) {
+    $tmp_folder = '/tmp/';
+
+    // Save the dbdump to a local destination.
+    if ('s3' === parse_url($filename, PHP_URL_SCHEME)) {
+      $command = sprintf(
+        'cd %s && ' .
+        's3cmd --force get %s',
+        $tmp_folder,
+        $filename
+      );
+
+      $shellProcess = $this->getShellProcess();
+      if ($shellProcess->exec($command, TRUE)) {
+        $this->io->writeln($shellProcess->getOutput());
+      }
+      else {
+        throw new SiteCommandException($shellProcess->getOutput());
+      }
+    }
+    else {
+      // Open a stream in read-only mode
+      if ($stream = fopen($filename, 'r')) {
```

Comment:
I think it would be a good idea to move the bit of logic to deal with copying the dump in execute() into download() and let it deal with the source (local, s3, etc)